distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso

```diff
@@ -619,3 +704,27 @@ truncate_table_truncate_table_style dialect table_name =
 default_fetch_types_query dialect expression context where_filter_always_false_literal="FALSE" =
     empty_context = context.add_where_filters [SQL_Expression.Literal where_filter_always_false_literal]
     dialect.generate_sql (Query.Select [["typed_column", expression]] empty_context)
+
+## PRIVATE
+   Helper class for shortening the binder names generated for WITH clauses.
+type Let_Binder_Renamer
+    Value (existing_tables : Hashset Text) (binders_ref : Ref Dictionary) (serial_ref : Ref Integer)
+
+    new (existing_tables : Vector Text) -> Let_Binder_Renamer =
+        Let_Binder_Renamer.Value (Hashset.from_vector existing_tables) (Ref.new Dictionary.empty) (Ref.new 0)
+
+    rename self (name : Text) (binder : Text) =
```

Comment:
I think the consensus is all methods should have `##` comment, probably with `PRIVATE` here.

Also I'd appreciate a short note on the logic here - I assume `binder` is the unique identifier and when we first encounter it, we try to find a shorter name based on `name` for it, and if we encounter the same identifier again we return the already assigned name?

Reply:
Exactly. I added docs.
Just curious -- if the type is PRIVATE, do the methods have to be PRIVATE too?

Reply:
Tbh I don't know, hence I add PRIVATE just to be sure.

I think they are not private, so if you somehow got a value of this private type, they probably would show up in CB on it. But that seems harmless as long as they do not show up in CB search 'in general' (so e.g. in the default mode where you have no value selected).

I think we just had this convention to mark methods as PRIVATE unless meant for users, but it was from time before we got `private` types so it's a question of how we want to amend that convention for that case.

Also, I'm not sure how this plays with @jdunkerley's code linter/formatter. I assume it may 'warn' about these non-PRIVATE functions, but again I'm not sure.